api/organisations/views.py

```diff
@@ -260,9 +265,87 @@ def my_permissions(self, request, pk):
         return Response(serializer.data)
 
 
+def chargebee_webhook_payment_failed(request: Request) -> Response:
+    if "content" not in request.data:
+        logger.warning("Missing `content` field from chargebee webhook")
+        return Response(status=status.HTTP_200_OK)
+
+    if "invoice" not in request.data["content"]:
+        logger.warning("Missing `invoice` field from chargebee webhook")
+        return Response(status=status.HTTP_200_OK)
+
+    invoice = request.data["content"]["invoice"]
+
+    # If dunning hasn't started ignore the declined payment.
+    if invoice.get("dunning_status") != "in_progress":
+        return Response(status=status.HTTP_200_OK)
+
+    if "subscription_id" not in invoice:
+        logger.info("Payment declined for non-subscription related charge, skipping.")
+        return Response(status=status.HTTP_200_OK)
```

Comment:
There are a lot of returns here (I have a feeling that flake8 may even complain about the complexity). Could we take the opportunity to start using some sort of serialization here? Perhaps a DRF serializer or just a marshmallow schema?

Reply:
I've switched it to a serializer locally, but it does come at the cost of detailed error messages. Even so, I've added stacktraces to the logged data so we should be ok. Will update this PR by tomorrow.